src/Raven.Server/ServerWide/ClusterStateMachine.cs

```diff
@@ -1472,54 +1472,69 @@ private void RemoveNodeFromDatabase(ClusterOperationContext context, BlittableJs
                 var databaseName = remove.DatabaseName;
                 var isShard = ShardHelper.TryGetShardNumberAndDatabaseName(databaseName, out var shardedDatabaseName, out var shardNumber);
                 var databaseNameLowered = shardedDatabaseName.ToLowerInvariant();
+
                 using (Slice.From(context.Allocator, "db/" + databaseNameLowered, out Slice lowerKey))
                 using (Slice.From(context.Allocator, "db/" + shardedDatabaseName, out Slice key))
                 {
                     var rawRecord = ReadRawDatabaseRecord(context, shardedDatabaseName, out _);
                     if (rawRecord == null)
                         throw new DatabaseDoesNotExistException($"The database {databaseName} does not exists");
+                    
+                    DatabaseRecord databaseRecord;
 
-                    DatabaseTopology topology;
-
-                    if (isShard)
+                    if (isShard) //shard database
                     {
-                        rawRecord.Sharding.Shards.TryGetValue(shardNumber, out topology);
+                        rawRecord.Sharding.Shards.TryGetValue(shardNumber, out var topology);
+
+                        if (topology == null)
```

Comment:
better to 
```
if(rawRecord.Sharding.Shards.TryGetValue(shardNumber, out var topology) == false)
   return;
```

Reply:
BTW where do you use the `topology` ?

Reply:
No where. Old code leftover. I guess It's enough to check `ContainsKey` here